course/src/isometry.cc

```diff
@@ -308,5 +334,96 @@ const Matrix3 Matrix3::kZero =
+    rot_result.r1().y() = (rotation_.r1().z() * rotation_.r3().y()) - (rotation_.r3().z() * rotation_.r1().y());
+    rot_result.r1().z() = (rotation_.r1().y() * rotation_.r2().z()) - (rotation_.r2().y() * rotation_.r1().z());
+
+    rot_result.r2().x() = (rotation_.r2().z() * rotation_.r3().x()) - (rotation_.r3().z() * rotation_.r2().x());
+    rot_result.r2().y() = (rotation_.r1().x() * rotation_.r3().z()) - (rotation_.r3().x() * rotation_.r1().z());
+    rot_result.r2().z() = (rotation_.r1().z() * rotation_.r2().x()) - (rotation_.r2().z() * rotation_.r1().x());
+
+    rot_result.r3().x() = (rotation_.r2().x() * rotation_.r3().y()) - (rotation_.r3().x() * rotation_.r2().y());
+    rot_result.r3().y() = (rotation_.r1().y() * rotation_.r3().x()) - (rotation_.r3().y() * rotation_.r1().x());
+    rot_result.r3().z() = (rotation_.r1().x() * rotation_.r2().y()) - (rotation_.r2().x() * rotation_.r1().y());
+
+    rot_result = rot_result / det;
+    auto vector_result = rot_result * translation_;
+    vector_result = vector_result * (-1);
+
+    return Isometry(vector_result, rot_result);
+}
+
+Isometry Isometry::FromTranslation(const std::initializer_list<double>& values) {
+    return Isometry(Vector3(values), Matrix3::kIdentity);
+}
+
+Isometry Isometry::FromEulerAngles(double yaw, double pitch, double roll) {
+    Isometry result = Isometry(RotateAround(Vector3::kUnitX, yaw)) * Isometry(RotateAround(Vector3::kUnitY, pitch)) *
+            Isometry(RotateAround(Vector3::kUnitZ, roll));
+    return result;
+}
+
+Matrix3 Isometry::RotateAround(const Vector3& axis, double angle) {
+    Matrix3 result;
+    auto cos = std::cos(angle);
+    auto sin = std::sin(angle);
+    result.r1().x() = cos + (axis.x() * axis.x()) * (1 - cos);
+    result.r1().y() = axis.x() * axis.y() * (1 - cos) - axis.z() * sin;
+    result.r1().z() = axis.x() * axis.z() * (1 - cos) + axis.y() * sin;
+
+    result.r2().x() = axis.y() * axis.x() * (1 - cos) + axis.z() * sin;
+    result.r2().y() = cos + (axis.y() * axis.y()) * (1 - cos);
+    result.r2().z() = axis.y() * axis.z() * (1 - cos) - axis.x() * sin;
+
+    result.r3().x() = axis.z() * axis.x() * (1 - cos) + axis.y() * sin;
+    result.r3().y() = axis.z() * axis.y() * (1 - cos) + axis.x() * sin;
+    result.r3().z() = cos + (axis.z() * axis.z()) * (1 - cos);
+    return result;
+}
+
+// // Operators
+bool Isometry::operator==(const Isometry& isometri) const {
+    if (rotation_ != isometri.rotation_) {
+        return false;
+    }
+    if (translation_ != isometri.translation_) {
+        return false;
+    }
+    return true;
+}
+
+Vector3 Isometry::operator*(const Vector3& vector) const{
+    return (rotation_ * vector + translation_);
+}
+
+Isometry Isometry::operator*(const Isometry& isometry) const {
+    Isometry aux(*this);
+    aux.rotation_ *= isometry.rotation();
+    aux.translation_ = (aux.rotation_ * isometry.translation()) + aux.translation_;
```

Comment:
This is correct, but it's a bit weird that you initialize `aux`, then overwrite it. Please create the return isometry with its final rotation and translation by operating with `isometry`'s rotation and translation, and the `rotation_` and `translation_` members of `*this`